control-base/src/main/java/fi/nls/oskari/control/layer/SaveLayerHandler.java

```diff
@@ -324,11 +324,13 @@ private boolean handleRequestToMapLayer(final ActionParameters params, OskariLay
             }
         }
 
-        String groupId = params.getHttpParam(PARAM_MAPLAYER_GROUPS, "-1");
-        ml.emptyMaplayerGroups();
-        for (String id: groupId.split(",")) {
-            MaplayerGroup maplayerGroup = oskariMapLayerGroupService.find(ConversionHelper.getInt(id, -1));
-            ml.addGroup(maplayerGroup);
+        String groupId = params.getHttpParam(PARAM_MAPLAYER_GROUPS);
+        if(groupId != null) {
+            ml.emptyMaplayerGroups();
+            for (String id: groupId.split(",")) {
+                MaplayerGroup maplayerGroup = oskariMapLayerGroupService.find(ConversionHelper.getInt(id, -1));
```

Comment:
Probably no point in calling the find() if the getInt() returns -1, but this code should get replaced pretty soon anyways.